lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3413,6 +3413,21 @@ void AttributeChecker::checkOriginalDefinedInAttrs(Decl *D,
   // Attrs are in the reverse order of the source order. We need to visit them
   // in source order to diagnose the later attribute.
   for (auto *Attr: Attrs) {
+    static StringRef AttrName = "_originallyDefinedIn";
+
+    if (auto *VD = dyn_cast<ValueDecl>(D)) {
+      // This attribute does not make sense on private/fileprivate declarations
+      // since clients can't use them. It probably also doesn't make sense for
+      // the majority of internal declarations, but those may be usable from
+      // inline or accessible externally when testing.
+      if (VD->getFormalAccess() <= AccessLevel::FilePrivate) {
```

Comment:
If you want to account for testable and the like, use `getFormalAccessScope()` instead.

Reply:
Maybe I'm missing something about the suggestion, but I'm having trouble imagining how it would work to account for `@testable`. If I'm building module `Foo` in isolation and it has an internal `@_originallyDefinedIn` declaration `Bar`, I don't know in that context whether there is test module out there somewhere that could potentially `@testable import Foo` and use `Bar`, right?

Reply:
Agreed. We should also specify `treatUsableFromInlineAsPublic=true` so that we don't warn on those symbols because they are effectively public.

Reply:
After discussing with @nkcsgexi I'm going to stop worrying about the `@testable` case as a situation where someone might want to use an internal symbol during testing and that symbol needs to be annotated `@_originallyDefinedIn` seems very contrived and unlikely to happen in practice, and the benefit from warning on `internal` declarations is large.